<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yunqi Yang" />


<title>Self-report asthma: Coxph-SuSIE</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">coxPHSuSIE-data-application</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/yunqiyang0215/coxPHSuSIE-data-application">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Self-report asthma: Coxph-SuSIE</h1>
<h4 class="author">Yunqi Yang</h4>
<h4 class="date">6/20/2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-06-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>survival-data-analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240324code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20240324)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240324code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240324)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215survivaldataanalysistreee5d84dd7238d4a04ec4d4133ce12fc5f76e152f1targetblanke5d84dda"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/yunqiyang0215/survival-data-analysis/tree/e5d84dd7238d4a04ec4d4133ce12fc5f76e152f1" target="_blank">e5d84dd</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215survivaldataanalysistreee5d84dd7238d4a04ec4d4133ce12fc5f76e152f1targetblanke5d84dda" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/yunqiyang0215/survival-data-analysis/tree/e5d84dd7238d4a04ec4d4133ce12fc5f76e152f1" target="_blank">e5d84dd</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Unstaged changes:
    Deleted:    analysis/logistic_gwas_asthma.Rmd
    Modified:   analysis/plot_km.Rmd
    Deleted:    analysis/susie_asthma_result.Rmd
    Modified:   analysis/susie_asthma_result1.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/susie_asthma_result2.Rmd</code>) and HTML (<code>docs/susie_asthma_result2.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-data-analysis/e5d84dd7238d4a04ec4d4133ce12fc5f76e152f1/docs/susie_asthma_result2.html" target="_blank">e5d84dd</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/ca1554fc022283dbba00df6c57d002068801a75f/analysis/susie_asthma_result2.Rmd" target="_blank">ca1554f</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-28
</td>
<td>
wflow_publish("analysis/susie_asthma_result2.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-data-analysis/7199ead69aa737bbf946eae707d2b3458ec43831/docs/susie_asthma_result2.html" target="_blank">7199ead</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/b8f7f428d4531216faf4b59821df02477b996aea/analysis/susie_asthma_result2.Rmd" target="_blank">b8f7f42</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-27
</td>
<td>
wflow_publish("analysis/susie_asthma_result2.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-data-analysis/b7003474a5621d0b2505affe593782361b83f730/docs/susie_asthma_result2.html" target="_blank">b700347</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/9148531c4d1d5b90e3e1addb9a73df3eaace323c/analysis/susie_asthma_result2.Rmd" target="_blank">9148531</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-27
</td>
<td>
wflow_publish("analysis/susie_asthma_result2.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-data-analysis/621141623786c2642958d28504556a44d59ac642/docs/susie_asthma_result2.html" target="_blank">6211416</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/87b8057fa7bde9791adfd4c922f9656bdfb11ef9/analysis/susie_asthma_result2.Rmd" target="_blank">87b8057</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-21
</td>
<td>
wflow_publish("analysis/susie_asthma_result2.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-data-analysis/aa454ea5a63841cc6d8517e335850317b03160c9/docs/susie_asthma_result2.html" target="_blank">aa454ea</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/5aa21af0949d3824d7f1e5801dfccc0b7b6fd7be/analysis/susie_asthma_result2.Rmd" target="_blank">5aa21af</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-21
</td>
<td>
wflow_publish("analysis/susie_asthma_result2.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-data-analysis/73f91cec9dd826d5a95372cd793d2e6d23b51952/docs/susie_asthma_result2.html" target="_blank">73f91ce</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/ed3b6e29fecad555956f6c848de5b56066654177/analysis/susie_asthma_result2.Rmd" target="_blank">ed3b6e2</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-20
</td>
<td>
wflow_publish("analysis/susie_asthma_result2.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-data-analysis/a4d10d11043b8b33fd2f7a6bb0c21f046e37c5e7/docs/susie_asthma_result2.html" target="_blank">a4d10d1</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/706dd1046589d76aa2bdf1e49ca3e19bde74971f/analysis/susie_asthma_result2.Rmd" target="_blank">706dd10</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-20
</td>
<td>
wflow_publish("analysis/susie_asthma_result2.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="description" class="section level2">
<h2>Description:</h2>
<p>Coxph Susie result on all asthma/ AOA/ COA in UKBiobank.</p>
<pre class="r"><code>library(survival)
library(susieR)
devtools::load_all(&quot;/Users/nicholeyang/Downloads/logisticsusie&quot;)</code></pre>
<pre><code>ℹ Loading logisticsusie</code></pre>
<div id="region-1" class="section level3">
<h3>Region 1</h3>
<p>Marginal significant signals for COA, weak signals for AOA.</p>
<p>rs11071559_T was the one with smallest pvalue in all asthma, and PIP = 0.24. Carole’s paper also reported this one as the top signal. But in AOA, it’s not the one with smallest pval, the pip is a lot smaller.</p>
<div id="all-asthma-cases" class="section level4">
<h4>1. All asthma cases</h4>
<pre class="r"><code>region = &quot;chr15_59000001_63400000&quot;
res = readRDS(paste0(&quot;/Users/nicholeyang/Downloads/survivalsusie/result/asthma_self_report/result/all/fit.susie.&quot;, region, &quot;.rds&quot;))
gwas = readRDS(paste0(&quot;/Users/nicholeyang/downloads/survivalsusie/result/gwas_surv/all_gwas_&quot;, region, &quot;.rds&quot;))</code></pre>
<pre class="r"><code>fit = res[[1]]
X = res[[2]]
print(res[[3]])</code></pre>
<pre><code>     user    system   elapsed 
48021.738 27994.163  4110.945 </code></pre>
<pre class="r"><code>pip &lt;- logisticsusie:::get_pip(fit$alpha)
effect_estimate &lt;- colSums(fit$alpha * fit$mu)
pip.sorted = sort(pip, decreasing = TRUE)
pip.sorted[1:10]</code></pre>
<pre><code> [1] 0.24346967 0.10867456 0.10424352 0.10165664 0.08854406 0.06504377
 [7] 0.05878656 0.04595972 0.04407223 0.04344761</code></pre>
<pre class="r"><code>class(fit) = &quot;susie&quot;
cs &lt;- susie_get_cs(fit, X)
cs</code></pre>
<pre><code>$cs
$cs$L1
 [1] 428 438 442 444 446 453 454 455 460 478 480 482 485 490 492 497 498 499 501


$purity
   min.abs.corr mean.abs.corr median.abs.corr
L1    0.7701707     0.9518411        0.965287

$cs_index
[1] 1

$coverage
[1] 0.9584117

$requested_coverage
[1] 0.95</code></pre>
<pre class="r"><code>snps1 = colnames(X)[cs$cs$L1]
colors &lt;- ifelse(rownames(gwas) %in% snps1, &quot;red&quot;, &quot;black&quot;)
plot(-log10(gwas[, &quot;p.value.spa&quot;]), col = colors, xlab = &quot;SNP&quot;, ylab = &quot;-log10(p-value)&quot;, cex = 0.8, pch = 20, main = &quot;CS 1&quot;)</code></pre>
<p><img src="figure/susie_asthma_result2.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/a4d10d11043b8b33fd2f7a6bb0c21f046e37c5e7/docs/figure/susie_asthma_result2.Rmd/unnamed-chunk-4-1.png" target="_blank">a4d10d1</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cbind(gwas[rownames(gwas) %in% snps1, ], pip[sort(cs$cs$L1)])</code></pre>
<pre><code>                   MAF missing.rate  p.value.spa p.value.norm      Stat
rs7183955_C  0.1857994            0 2.047135e-12 1.965395e-12 -686.9818
rs922783_G   0.1348981            0 7.063027e-12 6.654735e-12 -589.8232
rs12900122_T 0.1334614            0 2.805769e-12 2.621221e-12 -598.6492
rs12903966_T 0.1334992            0 2.479238e-12 2.313825e-12 -600.2511
rs16943087_G 0.1333576            0 4.033002e-12 3.778604e-12 -594.3751
rs2279294_C  0.1335081            0 1.268415e-11 1.199169e-11 -581.5727
rs2279293_G  0.1333651            0 1.466795e-11 1.388085e-11 -579.6246
rs2279292_C  0.1345950            0 5.626978e-12 5.290515e-12 -593.7446
rs8025689_C  0.1352865            0 8.350735e-12 7.879322e-12 -590.3153
rs12905602_A 0.1333781            0 7.234467e-12 6.809929e-12 -588.6392
rs11633029_C 0.1349144            0 1.688390e-11 1.601014e-11 -580.5839
rs11637671_G 0.1349347            0 1.629390e-11 1.544692e-11 -581.0378
rs11639084_T 0.1321305            0 1.409666e-11 1.332602e-11 -577.8021
rs10519067_A 0.1268612            0 2.796316e-12 2.598234e-12 -588.6569
rs10519068_A 0.1281122            0 1.097777e-12 1.012287e-12 -601.5405
rs11071557_C 0.1300621            0 1.018211e-12 9.399860e-13 -606.1218
rs34753162_C 0.1300892            0 9.385815e-13 8.658258e-13 -607.0330
rs34986765_C 0.1298710            0 1.199225e-12 1.108639e-12 -603.5777
rs11071559_T 0.1282230            0 4.231213e-13 3.864963e-13 -613.6394
                  Var         z           
rs7183955_C  9530.716 -7.036917 0.02401601
rs922783_G   7382.065 -6.864880 0.02646692
rs12900122_T 7320.864 -6.996668 0.05878656
rs12903966_T 7323.492 -7.014131 0.06504377
rs16943087_G 7324.006 -6.945224 0.04407223
rs2279294_C  7357.135 -6.780312 0.01930962
rs2279293_G  7353.770 -6.759145 0.01772081
rs2279292_C  7409.842 -6.897555 0.03527074
rs8025689_C  7446.700 -6.840725 0.02459909
rs12905602_A 7359.514 -6.861588 0.02795568
rs11633029_C 7423.555 -6.738435 0.01539183
rs11637671_G 7423.697 -6.743638 0.01567289
rs11639084_T 7294.842 -6.765053 0.01741284
rs10519067_A 7076.016 -6.997902 0.04595972
rs10519068_A 7120.226 -7.128826 0.10424352
rs11071557_C 7208.462 -7.139020 0.10165664
rs34753162_C 7207.339 -7.150309 0.10867456
rs34986765_C 7193.795 -7.116299 0.08854406
rs11071559_T 7143.786 -7.260208 0.24346967</code></pre>
<pre class="r"><code>rm(res, gwas, X, fit)</code></pre>
</div>
<div id="coa" class="section level4">
<h4>2. COA</h4>
<pre class="r"><code>region = &quot;chr15_59000001_63400000&quot;
res = readRDS(paste0(&quot;/Users/nicholeyang/Downloads/survivalsusie/result/asthma_self_report/result/coa/fit.susie.&quot;, region, &quot;.rds&quot;))
gwas = readRDS(paste0(&quot;/Users/nicholeyang/downloads/survivalsusie/result/gwas_surv/coa_gwas_&quot;, region, &quot;.rds&quot;))</code></pre>
<pre class="r"><code>fit = res[[1]]
X = res[[2]]

pip &lt;- logisticsusie:::get_pip(fit$alpha)
effect_estimate &lt;- colSums(fit$alpha * fit$mu)

class(fit) = &quot;susie&quot;
cs &lt;- susie_get_cs(fit, X)
cs</code></pre>
<pre><code>$cs
$cs$L1
 [1] 399 402 404 412 413 414 418 420 421 422 427 438 442 444 446 453 454 455 460
[20] 478 480 482 485 490 492 497 498 499 501


$purity
   min.abs.corr mean.abs.corr median.abs.corr
L1    0.9257579     0.9692791       0.9758275

$cs_index
[1] 1

$coverage
[1] 0.9594382

$requested_coverage
[1] 0.95</code></pre>
<pre class="r"><code>snps1 = colnames(X)[cs$cs$L1]
colors &lt;- ifelse(rownames(gwas) %in% snps1, &quot;red&quot;, &quot;black&quot;)
plot(-log10(gwas[, &quot;p.value.spa&quot;]), col = colors, xlab = &quot;SNP&quot;, ylab = &quot;-log10(p-value)&quot;, cex = 0.5, pch = 20, main = &quot;CS 1&quot;)</code></pre>
<p><img src="figure/susie_asthma_result2.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/a4d10d11043b8b33fd2f7a6bb0c21f046e37c5e7/docs/figure/susie_asthma_result2.Rmd/unnamed-chunk-9-1.png" target="_blank">a4d10d1</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cbind(gwas[rownames(gwas) %in% snps1, ], pip[sort(cs$cs$L1)])</code></pre>
<pre><code>                   MAF missing.rate  p.value.spa p.value.norm      Stat
rs1351544_T  0.1341295            0 4.461386e-10 3.713514e-10 -269.0891
rs1817479_C  0.1344601            0 3.808552e-10 3.158273e-10 -270.4024
rs8025324_A  0.1344176            0 3.725567e-10 3.087450e-10 -270.5693
rs16943064_A 0.1364784            0 5.377680e-10 4.518186e-10 -269.8478
rs9920526_T  0.1343697            0 3.189260e-10 2.631108e-10 -270.7226
rs9920610_C  0.1348054            0 4.116391e-10 3.422020e-10 -269.6961
rs9920560_A  0.1338591            0 5.835716e-10 4.889988e-10 -267.0437
rs9920592_T  0.1339444            0 5.579596e-10 4.670002e-10 -267.6067
rs9920593_T  0.1339534            0 5.553709e-10 4.647776e-10 -267.6369
rs1020730_T  0.1339552            0 5.392532e-10 4.509357e-10 -267.8550
rs7162065_A  0.1339407            0 4.841433e-10 4.036654e-10 -268.7056
rs922783_G   0.1351647            0 6.584186e-11 5.189163e-11 -283.4299
rs12900122_T 0.1337288            0 8.097954e-11 6.399853e-11 -280.9145
rs12903966_T 0.1337666            0 7.256387e-11 5.715275e-11 -281.6921
rs16943087_G 0.1336174            0 2.102903e-10 1.710490e-10 -274.5713
rs2279294_C  0.1337616            0 1.231272e-10 9.858842e-11 -278.7902
rs2279293_G  0.1336176            0 1.649137e-10 1.331822e-10 -276.7600
rs2279292_C  0.1348538            0 1.395559e-10 1.124347e-10 -278.9279
rs8025689_C  0.1355442            0 1.067680e-10 8.547625e-11 -281.4096
rs12905602_A 0.1336304            0 4.023675e-10 3.334537e-10 -270.7834
rs11633029_C 0.1351697            0 5.155685e-10 4.315266e-10 -270.2153
rs11637671_G 0.1351902            0 5.001905e-10 4.183294e-10 -270.4283
rs11639084_T 0.1323729            0 3.352193e-10 2.756701e-10 -270.8541
rs10519067_A 0.1271050            0 1.167764e-10 9.181477e-11 -273.8719
rs10519068_A 0.1283646            0 1.319142e-10 1.044645e-10 -273.9002
rs11071557_C 0.1303221            0 1.019793e-10 8.049745e-11 -277.2735
rs34753162_C 0.1303492            0 1.012173e-10 7.988084e-11 -277.3011
rs34986765_C 0.1301288            0 8.559017e-11 6.714989e-11 -278.1494
rs11071559_T 0.1284794            0 9.641085e-11 7.556718e-11 -276.4440
                  Var         z           
rs1351544_T  1844.440 -6.265617 0.02106368
rs1817479_C  1847.604 -6.290803 0.02338094
rs8025324_A  1847.817 -6.294323 0.02372920
rs16943064_A 1873.125 -6.234986 0.01829921
rs9920526_T  1835.440 -6.319088 0.02661092
rs9920610_C  1845.268 -6.278342 0.02221628
rs9920560_A  1841.711 -6.222595 0.01785386
rs9920592_T  1845.204 -6.229810 0.01827960
rs9920593_T  1845.178 -6.230558 0.01833191
rs1020730_T  1845.380 -6.235292 0.01864945
rs7162065_A  1846.851 -6.252604 0.01989128
rs922783_G   1863.667 -6.565405 0.09146328
rs12900122_T 1848.325 -6.534087 0.07795816
rs12903966_T 1848.987 -6.551001 0.08590416
rs16943087_G 1849.039 -6.385311 0.03510412
rs2279294_C  1857.237 -6.469100 0.05450854
rs2279293_G  1856.368 -6.423493 0.04278518
rs2279292_C  1870.557 -6.449210 0.04876977
rs8025689_C  1879.766 -6.490633 0.06000053
rs12905602_A 1857.793 -6.282368 0.02189398
rs11633029_C 1873.906 -6.242175 0.01838518
rs11637671_G 1873.945 -6.247031 0.01872130
rs11639084_T 1841.423 -6.311878 0.02482659
rs10519067_A 1786.345 -6.479848 0.05514732
rs10519068_A 1797.518 -6.460345 0.04934584
rs11071557_C 1819.845 -6.499668 0.05895974
rs34753162_C 1819.559 -6.500825 0.05941099
rs34986765_C 1816.117 -6.526889 0.06868704
rs11071559_T 1803.695 -6.509170 0.06395982</code></pre>
<pre class="r"><code>rm(res, gwas, X, fit)</code></pre>
</div>
<div id="aoa" class="section level4">
<h4>3. AOA</h4>
<pre class="r"><code>region = &quot;chr15_59000001_63400000&quot;
res = readRDS(paste0(&quot;/Users/nicholeyang/Downloads/survivalsusie/result/asthma_self_report/result/aoa/fit.susie.&quot;, region, &quot;.rds&quot;))
gwas = readRDS(paste0(&quot;/Users/nicholeyang/downloads/survivalsusie/result/gwas_surv/aoa_gwas_&quot;, region, &quot;.rds&quot;))</code></pre>
<pre class="r"><code>fit = res[[1]]
X = res[[2]]

pip &lt;- logisticsusie:::get_pip(fit$alpha)
effect_estimate &lt;- colSums(fit$alpha * fit$mu)

class(fit) = &quot;susie&quot;
cs &lt;- susie_get_cs(fit, X)
cs</code></pre>
<pre><code>$cs
NULL

$coverage
NULL

$requested_coverage
[1] 0.95</code></pre>
<pre class="r"><code>rm(res, gwas, X, fit)</code></pre>
</div>
</div>
<div id="region-2" class="section level3">
<h3>Region 2</h3>
<p>Very significant signals for COA, marginal significant signals for AOA.</p>
<p>All asthma has a very weird CS. One All asthma CS overlap with COA CS. AOA CS has no overlap with other CSs.</p>
<div id="all-asthma-cases-1" class="section level4">
<h4>1. All asthma cases</h4>
<pre class="r"><code>region = &quot;chr2_102100001_105300000&quot;
res = readRDS(paste0(&quot;/Users/nicholeyang/Downloads/survivalsusie/result/asthma_self_report/result/all/fit.susie.&quot;, region, &quot;.rds&quot;))
gwas = readRDS(paste0(&quot;/Users/nicholeyang/downloads/survivalsusie/result/gwas_surv/all_gwas_&quot;, region, &quot;.rds&quot;))</code></pre>
<pre class="r"><code>fit = res[[1]]
X = res[[2]]
print(res[[3]])</code></pre>
<pre><code>     user    system   elapsed 
370760.90 210483.81  30824.19 </code></pre>
<pre class="r"><code>pip &lt;- logisticsusie:::get_pip(fit$alpha)
effect_estimate &lt;- colSums(fit$alpha * fit$mu)
pip.sorted = sort(pip, decreasing = TRUE)
pip.sorted[1:10]</code></pre>
<pre><code> [1] 0.45753680 0.05677487 0.05635246 0.05510944 0.05506850 0.05371545
 [7] 0.04598294 0.04059181 0.03899255 0.03087550</code></pre>
<pre class="r"><code>class(fit) = &quot;susie&quot;
cs &lt;- susie_get_cs(fit, X)
cs</code></pre>
<pre><code>$cs
$cs$L2
 [1] 2205 2210 2212 2213 2214 2216 2220 2223 2224 2226 2233 2237 2238 2240 2241
[16] 2247 2248 2250 2251 2254 2255 2256 2257 2258 2260 2262 2264 2265 2266 2270
[31] 2273 2274 2276 2277 2279 2281 2282 2283 2287 2292 2295 2296 2299 2301 2302
[46] 2306 2307 2309 2311 2317 2328 2329 2330 2331 2332 2333 2336 2342 2344 2350
[61] 2352

$cs$L1
 [1] 2261 2267 2275 2280 2284 2285 2300 2304 2314 2323 2324 2362 2365 2367 2368
[16] 2369 2375 2384 2409


$purity
   min.abs.corr mean.abs.corr median.abs.corr
L2    0.9919537     0.9983229       0.9987347
L1    0.9862647     0.9966765       0.9973417

$cs_index
[1] 2 1

$coverage
[1] 0.9524084 0.9600456

$requested_coverage
[1] 0.95</code></pre>
<pre class="r"><code>par(mfrow = c(1,2))
snps1 = colnames(X)[cs$cs$L1]
colors &lt;- ifelse(rownames(gwas) %in% snps1, &quot;red&quot;, &quot;black&quot;)
plot(-log10(gwas[, &quot;p.value.spa&quot;]), col = colors, xlab = &quot;SNP&quot;, ylab = &quot;-log10(p-value)&quot;, cex = 0.8, pch = 20, main = &quot;CS 1&quot;)

snps2 = colnames(X)[cs$cs$L2]
colors &lt;- ifelse(rownames(gwas) %in% snps2, &quot;red&quot;, &quot;black&quot;)
plot(-log10(gwas[, &quot;p.value.spa&quot;]), col = colors, xlab = &quot;SNP&quot;, ylab = &quot;-log10(p-value)&quot;, cex = 0.8, pch = 20, main = &quot;CS 2&quot;)</code></pre>
<p><img src="figure/susie_asthma_result2.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/aa454ea5a63841cc6d8517e335850317b03160c9/docs/figure/susie_asthma_result2.Rmd/unnamed-chunk-17-1.png" target="_blank">aa454ea</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cbind(gwas[rownames(gwas) %in% snps1, ], pip[sort(cs$cs$L1)])</code></pre>
<pre><code>                   MAF missing.rate  p.value.spa p.value.norm      Stat
rs72823635_C 0.1380777            0 9.576229e-41 1.324452e-41 -1176.896
rs950881_T   0.1380726            0 9.748399e-41 1.349239e-41 -1176.738
rs10179458_T 0.1381013            0 7.403775e-41 1.012584e-41 -1178.446
rs72823641_A 0.1373666            0 4.732089e-42 5.654617e-43 -1193.554
rs10189154_T 0.1380466            0 6.148579e-41 8.334263e-42 -1179.481
rs10189526_T 0.1380464            0 6.114220e-41 8.285568e-42 -1179.512
rs11679893_A 0.1380527            0 6.375127e-41 8.656649e-42 -1179.217
rs10865050_A 0.1380920            0 5.368975e-41 7.237633e-42 -1180.414
rs12053429_T 0.1380637            0 6.694631e-41 9.108477e-42 -1178.781
rs59185885_G 0.1353869            0 2.241854e-41 2.845097e-42 -1167.877
rs58815545_T 0.1380667            0 6.716752e-41 9.140149e-42 -1178.782
rs3771180_T  0.1381264            0 5.608771e-41 7.588367e-42 -1180.484
rs72823646_A 0.1380350            0 3.541638e-41 4.678987e-42 -1183.517
rs13431828_T 0.1380081            0 3.139351e-41 4.123458e-42 -1184.197
rs13408569_C 0.1377062            0 2.707921e-41 3.529341e-42 -1182.912
rs13408661_A 0.1379078            0 2.726747e-41 3.557209e-42 -1184.521
rs10173081_T 0.1380032            0 2.770821e-41 3.618302e-42 -1184.931
rs3771175_A  0.1377430            0 5.047224e-41 6.738070e-42 -1180.195
rs10197862_G 0.1383215            0 5.426446e-41 7.337032e-42 -1181.441
                  Var         z           
rs72823635_C 7586.161 -13.51223 0.01274015
rs950881_T   7585.660 -13.51087 0.01250895
rs10179458_T 7583.970 -13.53198 0.01756912
rs72823641_A 7543.225 -13.74243 0.45753680
rs10189154_T 7581.257 -13.54628 0.02173272
rs10189526_T 7581.182 -13.54672 0.02188418
rs11679893_A 7580.990 -13.54350 0.02098567
rs10865050_A 7581.669 -13.55664 0.02550848
rs12053429_T 7579.567 -13.53976 0.01950383
rs59185885_G 7347.221 -13.62497 0.05677487
rs58815545_T 7579.865 -13.53951 0.01943143
rs3771180_T  7586.448 -13.55317 0.02007143
rs72823646_A 7585.757 -13.58861 0.03899255
rs13431828_T 7584.152 -13.59786 0.04598294
rs13408569_C 7555.054 -13.60923 0.05506850
rs13408661_A 7576.263 -13.60866 0.05510944
rs10173081_T 7582.898 -13.60741 0.05371545
rs3771175_A  7572.989 -13.56189 0.01948692
rs10197862_G 7595.980 -13.55564 0.01154937</code></pre>
<pre class="r"><code>cbind(gwas[rownames(gwas) %in% snps2, ], pip[sort(cs$cs$L2)])</code></pre>
<pre><code>                    MAF missing.rate p.value.spa p.value.norm      Stat
rs1420091_C   0.4741893            0  0.04343415   0.04343426 -254.3545
rs4399750_C   0.4742024            0  0.04219033   0.04219043 -255.9176
rs2110660_G   0.4739445            0  0.04202196   0.04202206 -256.1081
rs1420090_C   0.4741980            0  0.04195790   0.04195801 -256.2307
rs7565653_A   0.4741983            0  0.04197044   0.04197054 -256.2175
rs7568913_C   0.4726704            0  0.03376352   0.03376353 -267.3910
rs10179654_G  0.4725454            0  0.03006019   0.03006016 -273.1701
rs4090473_G   0.4741994            0  0.04287977   0.04287988 -255.1204
rs12476925_T  0.4738162            0  0.04398899   0.04398911 -253.7171
rs12476968_T  0.4738706            0  0.04252796   0.04252807 -255.4688
rs6721346_C   0.4737530            0  0.04447458   0.04447471 -253.1289
rs10178436_C  0.4726547            0  0.03684838   0.03684842 -262.9061
rs11679191_T  0.4737891            0  0.04481519   0.04481532 -252.7314
rs11685424_A  0.4737905            0  0.04494659   0.04494673 -252.5758
rs11685480_A  0.4743119            0  0.04743351   0.04743351 -249.7420
rs6733174_C   0.4738143            0  0.04755366   0.04755366 -249.5822
rs6543118_A   0.4727228            0  0.04026244   0.04026251 -258.3334
rs1558622_A   0.4742182            0  0.04659919   0.04659919 -250.6703
rs1558621_G   0.4738770            0  0.04687411   0.04687411 -250.3313
rs10189202_G  0.4741925            0  0.04793089   0.04793089 -249.1883
rs10191914_C  0.4742151            0  0.04757686   0.04757686 -249.5846
rs10189711_G  0.4738113            0  0.04764314   0.04764314 -249.4869
rs12712135_G  0.4726901            0  0.03887182   0.03887189 -260.1599
rs1558620_C   0.4742148            0  0.04802466   0.04802466 -249.0875
rs1558619_T   0.4740595            0  0.04795157   0.04795157 -249.1596
rs12996505_G  0.4730879            0  0.03883659   0.03883666 -260.2388
rs13020793_T  0.4742242            0  0.04813716   0.04813716 -248.9577
rs10183388_T  0.4730986            0  0.03983078   0.03983086 -258.9293
rs953934_T    0.4729525            0  0.04103713   0.04103721 -257.3664
rs1968171_T   0.4742123            0  0.04794220   0.04794220 -249.1836
rs4613307_G   0.4742125            0  0.04796262   0.04796262 -249.1609
rs1968170_A   0.4742026            0  0.04758388   0.04758388 -249.5872
rs11123918_C  0.4741587            0  0.05052438   0.05052438 -246.3781
rs10182639_A  0.4743496            0  0.05203128   0.05203128 -244.7823
rs11690443_A  0.4744681            0  0.05095669   0.05095669 -245.9098
rs12712136_C  0.4744596            0  0.05064872   0.05064872 -246.2421
rs974389_A    0.4740822            0  0.05216000   0.05216000 -244.6285
rs4142132_A   0.4744654            0  0.05064178   0.05064178 -246.2548
rs971764_T    0.4739114            0  0.05168196   0.05168196 -245.1564
rs1420088_C   0.4743646            0  0.04918105   0.04918105 -247.8470
rs11123920_T  0.4743140            0  0.04989757   0.04989757 -247.0801
rs6706844_C   0.4739374            0  0.05133120   0.05133120 -245.5290
rs11675988_C  0.4762069            0  0.05698309   0.05698309 -239.4453
rs11679900_T  0.4745015            0  0.05533114   0.05533114 -241.2578
rs11676075_C  0.4742919            0  0.05031902   0.05031902 -246.6278
rs11123921_G  0.4742925            0  0.05032595   0.05032595 -246.6205
rs12992762_C  0.4743465            0  0.05600560   0.05600560 -240.5810
rs12998412_C  0.4745086            0  0.05516205   0.05516205 -241.4267
rs11123922_C  0.4745087            0  0.05516879   0.05516879 -241.4203
rs12725988_T  0.4742123            0  0.05333701   0.05333701 -243.1924
rs76520363_A  0.4743312            0  0.05234657   0.05234657 -244.4862
rs76278109_G  0.4743494            0  0.05592508   0.05592508 -240.6635
rs76886731_T  0.4746917            0  0.05296045   0.05296045 -243.5965
rs150341880_T 0.4746447            0  0.05611893   0.05611893 -240.4858
rs138087973_G 0.4745021            0  0.05349105   0.05349105 -243.1116
rs76498201_G  0.4745064            0  0.05347515   0.05347515 -243.1284
rs12996772_T  0.4742928            0  0.05040735   0.05040735 -246.5351
rs1420102_T   0.4741610            0  0.05231199   0.05231199 -244.5174
rs12466380_G  0.4742917            0  0.05051025   0.05051025 -246.4174
rs1997467_G   0.4743260            0  0.04768922   0.04768922 -249.4460
rs1997466_G   0.4743715            0  0.05098088   0.05098088 -245.9699
                   Var         z            
rs1420091_C   15863.06 -2.019510 0.020567980
rs4399750_C   15867.49 -2.031637 0.022447600
rs2110660_G   15865.11 -2.033302 0.022180323
rs1420090_C   15870.38 -2.033936 0.022911552
rs7565653_A   15870.70 -2.033812 0.022885254
rs7568913_C   15865.02 -2.122885 0.040591813
rs10179654_G  15857.26 -2.169297 0.056352463
rs4090473_G   15874.23 -2.024878 0.021416219
rs12476925_T  15867.06 -2.014195 0.018767742
rs12476968_T  15863.71 -2.028315 0.021014149
rs6721346_C   15866.06 -2.009589 0.018115441
rs10178436_C  15862.57 -2.087440 0.030875498
rs11679191_T  15866.84 -2.006384 0.017834895
rs11685424_A  15866.78 -2.005153 0.017675726
rs11685480_A  15870.72 -1.982409 0.015798015
rs6733174_C   15867.59 -1.981336 0.014815661
rs6543118_A   15863.92 -2.051046 0.023503809
rs1558622_A   15868.37 -1.989925 0.016641187
rs1558621_G   15865.14 -1.987436 0.015141082
rs10189202_G  15871.25 -1.977981 0.015177251
rs10191914_C  15871.21 -1.981129 0.015520890
rs10189711_G  15868.26 -1.980538 0.014732695
rs12712135_G  15864.02 -2.065541 0.026154895
rs1558620_C   15871.75 -1.977150 0.015064340
rs1558619_T   15870.54 -1.977797 0.014930782
rs12996505_G  15867.91 -2.065913 0.027628580
rs13020793_T  15871.17 -1.976156 0.014872891
rs10183388_T  15868.21 -2.055499 0.025444259
rs953934_T    15867.24 -2.043154 0.022750435
rs1968171_T   15872.26 -1.977880 0.015170485
rs4613307_G   15872.28 -1.977699 0.015149208
rs1968170_A   15872.55 -1.981066 0.015516929
rs11123918_C  15874.12 -1.955497 0.012995327
rs10182639_A  15873.42 -1.942875 0.012086474
rs11690443_A  15873.08 -1.951844 0.013046980
rs12712136_C  15873.70 -1.954444 0.013324062
rs974389_A    15870.85 -1.941811 0.011567172
rs4142132_A   15874.38 -1.954503 0.013322996
rs971764_T    15874.57 -1.945773 0.011583486
rs1420088_C   15876.33 -1.967019 0.014296830
rs11123920_T  15877.82 -1.960841 0.013555044
rs6706844_C   15875.06 -1.948701 0.011960113
rs11675988_C  15824.65 -1.903441 0.011531797
rs11679900_T  15850.78 -1.916267 0.010290338
rs11676075_C  15877.97 -1.957242 0.013213798
rs11123921_G  15877.98 -1.957183 0.013208009
rs12992762_C  15849.11 -1.910992 0.009792043
rs12998412_C  15850.96 -1.917598 0.010402649
rs11123922_C  15851.01 -1.917544 0.010396758
rs12725988_T  15841.77 -1.932182 0.010272990
rs76520363_A  15877.54 -1.940273 0.011192302
rs76278109_G  15849.57 -1.911619 0.009830752
rs76886731_T  15844.22 -1.935243 0.011286261
rs150341880_T 15851.18 -1.910111 0.009961313
rs138087973_G 15851.71 -1.930935 0.011417901
rs76498201_G  15851.78 -1.931064 0.011446676
rs12996772_T  15878.23 -1.956491 0.013134177
rs1420102_T   15876.94 -1.940557 0.011449107
rs12466380_G  15877.24 -1.955617 0.012973333
rs1997467_G   15869.63 -1.980127 0.014645578
rs1997466_G   15884.16 -1.951641 0.012393843</code></pre>
</div>
<div id="conditional-analysis" class="section level4">
<h4>2. Conditional analysis</h4>
<pre class="r"><code>gwas1 = readRDS(&quot;/Users/nicholeyang/downloads/survivalsusie/result/asthma_self_report/result/gwas_surv_conditional/all_gwas_chr2_102100001_105300000_rs72823641_A.rds&quot;)</code></pre>
<pre class="r"><code>snps2 = colnames(X)[cs$cs$L2]
colors &lt;- ifelse(rownames(gwas1) %in% snps2, &quot;red&quot;, &quot;black&quot;)
plot(-log10(gwas1[, &quot;p.value.spa&quot;]), col = colors, xlab = &quot;SNP&quot;, ylab = &quot;-log10(p-value)&quot;, cex = 0.8, pch = 20, main = &quot;CS 2&quot;)</code></pre>
<p><img src="figure/susie_asthma_result2.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/e5d84dd7238d4a04ec4d4133ce12fc5f76e152f1/docs/figure/susie_asthma_result2.Rmd/unnamed-chunk-20-1.png" target="_blank">e5d84dd</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>rm(res, gwas, X, fit)</code></pre>
</div>
<div id="coa-1" class="section level4">
<h4>2. COA</h4>
<pre class="r"><code>region = &quot;chr2_102100001_105300000&quot;
res = readRDS(paste0(&quot;/Users/nicholeyang/Downloads/survivalsusie/result/asthma_self_report/result/coa/fit.susie.&quot;, region, &quot;.rds&quot;))
gwas = readRDS(paste0(&quot;/Users/nicholeyang/downloads/survivalsusie/result/gwas_surv/coa_gwas_&quot;, region, &quot;.rds&quot;))</code></pre>
<pre class="r"><code>fit = res[[1]]
X = res[[2]]

pip &lt;- logisticsusie:::get_pip(fit$alpha)
effect_estimate &lt;- colSums(fit$alpha * fit$mu)

class(fit) = &quot;susie&quot;
cs &lt;- susie_get_cs(fit, X)
cs</code></pre>
<pre><code>$cs
$cs$L1
 [1] 2261 2267 2275 2280 2284 2285 2300 2304 2314 2318 2323 2324 2362 2365 2367
[16] 2368 2369 2375 2384 2409 2441 2469 2496


$purity
   min.abs.corr mean.abs.corr median.abs.corr
L1    0.9808287      0.993961       0.9959249

$cs_index
[1] 1

$coverage
[1] 0.9571007

$requested_coverage
[1] 0.95</code></pre>
<pre class="r"><code>snps1 = colnames(X)[cs$cs$L1]
colors &lt;- ifelse(rownames(gwas) %in% snps1, &quot;red&quot;, &quot;black&quot;)
plot(-log10(gwas[, &quot;p.value.spa&quot;]), col = colors, xlab = &quot;SNP&quot;, ylab = &quot;-log10(p-value)&quot;, cex = 0.5, pch = 20, main = &quot;CS 1&quot;)</code></pre>
<p><img src="figure/susie_asthma_result2.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/e5d84dd7238d4a04ec4d4133ce12fc5f76e152f1/docs/figure/susie_asthma_result2.Rmd/unnamed-chunk-24-1.png" target="_blank">e5d84dd</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cbind(gwas[rownames(gwas) %in% snps1, ], pip[sort(cs$cs$L1)])</code></pre>
<pre><code>                    MAF missing.rate  p.value.spa p.value.norm      Stat
rs72823635_C  0.1384102            0 3.640657e-36 3.586463e-38 -565.3715
rs950881_T    0.1384052            0 3.650732e-36 3.596916e-38 -565.3431
rs10179458_T  0.1384360            0 3.414756e-36 3.344086e-38 -565.5294
rs72823641_A  0.1377047            0 5.234953e-37 4.132563e-39 -571.0181
rs10189154_T  0.1383818            0 2.548608e-36 2.419064e-38 -566.5161
rs10189526_T  0.1383816            0 2.532777e-36 2.402474e-38 -566.5365
rs11679893_A  0.1383882            0 2.425671e-36 2.291413e-38 -566.6889
rs10865050_A  0.1384279            0 2.126636e-36 1.983525e-38 -567.2004
rs12053429_T  0.1383980            0 2.452400e-36 2.318132e-38 -566.5963
rs4625927_C   0.1360328            0 3.103566e-36 2.879335e-38 -558.4277
rs59185885_G  0.1357253            0 3.083702e-36 2.866879e-38 -557.1747
rs58815545_T  0.1384010            0 2.441741e-36 2.307155e-38 -566.6235
rs3771180_T   0.1384616            0 4.257845e-37 3.374196e-39 -573.2972
rs72823646_A  0.1383742            0 7.893597e-37 6.626818e-39 -571.0281
rs13431828_T  0.1383453            0 1.069497e-36 9.257254e-39 -569.8501
rs13408569_C  0.1380420            0 6.873934e-37 5.670846e-39 -570.3877
rs13408661_A  0.1382438            0 6.675565e-37 5.497518e-39 -571.2896
rs10173081_T  0.1383423            0 9.488544e-37 8.108849e-39 -570.2477
rs3771175_A   0.1380877            0 1.381360e-36 1.215203e-38 -568.5501
rs10197862_G  0.1386554            0 8.108652e-37 6.876994e-39 -571.2970
rs145573519_T 0.1386368            0 1.596594e-36 1.459462e-38 -564.9405
rs56179005_A  0.1386146            0 1.019437e-36 8.876248e-39 -570.0474
rs72823669_T  0.1384047            0 4.251175e-37 3.357369e-39 -572.6834
                   Var         z           
rs72823635_C  1915.629 -12.91749 0.01062986
rs950881_T    1915.503 -12.91727 0.01059881
rs10179458_T  1915.103 -12.92287 0.01146301
rs72823641_A  1905.031 -13.08274 0.10723183
rs10189154_T  1914.411 -12.94776 0.01590902
rs10189526_T  1914.392 -12.94829 0.01602472
rs11679893_A  1914.348 -12.95192 0.01687270
rs10865050_A  1914.531 -12.96299 0.01966084
rs12053429_T  1913.985 -12.95103 0.01665957
rs4625927_C   1863.986 -12.93438 0.01394483
rs59185885_G  1855.535 -12.93471 0.01386645
rs58815545_T  1914.062 -12.95140 0.01674456
rs3771180_T   1915.756 -13.09814 0.13722082
rs72823646_A  1915.608 -13.04681 0.06482151
rs13431828_T  1915.191 -13.02131 0.04476226
rs13408569_C  1907.841 -13.05867 0.07789837
rs13408661_A  1913.187 -13.06104 0.08058440
rs10173081_T  1914.890 -13.03142 0.05191360
rs3771175_A   1912.565 -13.00052 0.03332018
rs10197862_G  1918.242 -13.04398 0.05188656
rs145573519_T 1892.435 -12.98650 0.02369409
rs56179005_A  1915.573 -13.02452 0.03957983
rs72823669_T  1911.546 -13.09852 0.11447487</code></pre>
<pre class="r"><code>rm(res, gwas, X, fit)</code></pre>
</div>
<div id="aoa-1" class="section level4">
<h4>3. AOA</h4>
<pre class="r"><code>region = &quot;chr2_102100001_105300000&quot;
res = readRDS(paste0(&quot;/Users/nicholeyang/Downloads/survivalsusie/result/asthma_self_report/result/aoa/fit.susie.&quot;, region, &quot;.rds&quot;))
gwas = readRDS(paste0(&quot;/Users/nicholeyang/downloads/survivalsusie/result/gwas_surv/aoa_gwas_&quot;, region, &quot;.rds&quot;))</code></pre>
<pre class="r"><code>fit = res[[1]]
X = res[[2]]

pip &lt;- logisticsusie:::get_pip(fit$alpha)
effect_estimate &lt;- colSums(fit$alpha * fit$mu)

class(fit) = &quot;susie&quot;
cs &lt;- susie_get_cs(fit, X)
cs</code></pre>
<pre><code>$cs
$cs$L1
 [1] 2234 2263 2268 2291 2340 2345 2348 2377 2386 2388 2397 2401 2422


$purity
   min.abs.corr mean.abs.corr median.abs.corr
L1    0.9237923     0.9613092       0.9408626

$cs_index
[1] 1

$coverage
[1] 0.9553805

$requested_coverage
[1] 0.95</code></pre>
<pre class="r"><code>snps1 = colnames(X)[cs$cs$L1]
colors &lt;- ifelse(rownames(gwas) %in% snps1, &quot;red&quot;, &quot;black&quot;)
plot(-log10(gwas[, &quot;p.value.spa&quot;]), col = colors, xlab = &quot;SNP&quot;, ylab = &quot;-log10(p-value)&quot;, cex = 0.8, pch = 20, main = &quot;CS 1&quot;)</code></pre>
<p><img src="figure/susie_asthma_result2.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/e5d84dd7238d4a04ec4d4133ce12fc5f76e152f1/docs/figure/susie_asthma_result2.Rmd/unnamed-chunk-29-1.png" target="_blank">e5d84dd</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cbind(gwas[rownames(gwas) %in% snps1, ], pip[sort(cs$cs$L1)])</code></pre>
<pre><code>                   MAF missing.rate  p.value.spa p.value.norm     Stat      Var
rs12470864_A 0.3863606            0 3.271588e-13 3.219111e-13 682.5887 8779.527
rs13020553_G 0.3862615            0 3.197035e-13 3.145611e-13 682.9466 8781.230
rs950880_A   0.3863995            0 3.257407e-13 3.205116e-13 682.9326 8786.959
rs13001325_T 0.3861511            0 2.619143e-13 2.576125e-13 685.5612 8783.714
rs1420104_A  0.3861597            0 2.750791e-13 2.705829e-13 684.9582 8784.109
rs12479210_T 0.3864228            0 3.037822e-13 2.988743e-13 683.8086 8786.786
rs13019081_C 0.3862194            0 3.119249e-13 3.068887e-13 683.4134 8785.215
rs1420101_T  0.3817173            0 6.321158e-13 6.221672e-13 673.4475 8759.543
rs13001714_G 0.3936798            0 1.126760e-11 1.114961e-11 637.9647 8825.688
rs12712142_A 0.3936688            0 1.068219e-11 1.056966e-11 638.6946 8825.863
rs6543119_T  0.3934532            0 1.292892e-11 1.279536e-11 636.2281 8829.345
rs13017455_T 0.3934435            0 1.197511e-11 1.185034e-11 637.2591 8829.038
rs11123923_A 0.3936404            0 1.173727e-11 1.161501e-11 637.7911 8836.236
                    z            
rs12470864_A 7.284899 0.114354967
rs13020553_G 7.288012 0.116707204
rs950880_A   7.285486 0.114628070
rs13001325_T 7.314879 0.139194856
rs1420104_A  7.308280 0.133206898
rs12479210_T 7.294903 0.121686457
rs13019081_C 7.291339 0.119155594
rs1420101_T  7.195534 0.073043494
rs13001714_G 6.790822 0.007546732
rs12712142_A 6.798524 0.007857954
rs6543119_T  6.770934 0.006868286
rs13017455_T 6.782024 0.007266150
rs11123923_A 6.784921 0.007488015</code></pre>
<pre class="r"><code>rm(res, gwas, X, fit)</code></pre>
</div>
</div>
<div id="region-3" class="section level3">
<h3>Region 3</h3>
<p>No significant signals for COA, marginal significant signals for AOA.</p>
<div id="all-asthma-cases-2" class="section level4">
<h4>1. All asthma cases</h4>
<pre class="r"><code>region = &quot;chr2_143400001_147900000&quot;
res = readRDS(paste0(&quot;/Users/nicholeyang/Downloads/survivalsusie/result/asthma_self_report/result/all/fit.susie.&quot;, region, &quot;.rds&quot;))
gwas = readRDS(paste0(&quot;/Users/nicholeyang/downloads/survivalsusie/result/gwas_surv/all_gwas_&quot;, region, &quot;.rds&quot;))</code></pre>
<pre class="r"><code>fit = res[[1]]
X = res[[2]]
print(res[[3]])</code></pre>
<pre><code>    user   system  elapsed 
71863.82 24439.76 11045.23 </code></pre>
<pre class="r"><code>pip &lt;- logisticsusie:::get_pip(fit$alpha)
effect_estimate &lt;- colSums(fit$alpha * fit$mu)
pip.sorted = sort(pip, decreasing = TRUE)
pip.sorted[1:10]</code></pre>
<pre><code> [1] 0.04544534 0.04473008 0.04449596 0.04117635 0.04089321 0.04065538
 [7] 0.04018651 0.03991979 0.03925386 0.03923206</code></pre>
<pre class="r"><code>class(fit) = &quot;susie&quot;
cs &lt;- susie_get_cs(fit, X)
cs</code></pre>
<pre><code>$cs
$cs$L1
 [1] 815 817 820 828 831 832 848 849 850 852 854 855 869 871 872 875 876 877 881
[20] 902 911 918 920 922 931 937 951 952 954 958 961 966 970


$purity
   min.abs.corr mean.abs.corr median.abs.corr
L1    0.9331464     0.9711805        0.989993

$cs_index
[1] 1

$coverage
[1] 0.9564943

$requested_coverage
[1] 0.95</code></pre>
<pre class="r"><code>snps1 = colnames(X)[cs$cs$L1]
colors &lt;- ifelse(rownames(gwas) %in% snps1, &quot;red&quot;, &quot;black&quot;)
plot(-log10(gwas[, &quot;p.value.spa&quot;]), col = colors, xlab = &quot;SNP&quot;, ylab = &quot;-log10(p-value)&quot;, cex = 0.8, pch = 20, main = &quot;CS 1&quot;)</code></pre>
<p><img src="figure/susie_asthma_result2.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-34-1">
Past versions of unnamed-chunk-34-1.png
</button>
</p>
<div id="fig-unnamed-chunk-34-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/e5d84dd7238d4a04ec4d4133ce12fc5f76e152f1/docs/figure/susie_asthma_result2.Rmd/unnamed-chunk-34-1.png" target="_blank">e5d84dd</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cbind(gwas[rownames(gwas) %in% snps1, ], pip[sort(cs$cs$L1)])</code></pre>
<pre><code>                   MAF missing.rate  p.value.spa p.value.norm      Stat
rs961414_T   0.4668426            0 1.181296e-05 1.180891e-05 -550.9762
rs7596680_G  0.4668163            0 1.153933e-05 1.153535e-05 -551.6466
rs4422110_T  0.4668991            0 1.226399e-05 1.225982e-05 -550.0536
rs1474011_A  0.4621279            0 6.734099e-06 6.731454e-06 -566.5605
rs1533426_A  0.4618248            0 6.133450e-06 6.130990e-06 -568.9658
rs6430080_A  0.4617999            0 6.167679e-06 6.165208e-06 -568.8491
rs13009175_T 0.4620989            0 6.926270e-06 6.923567e-06 -565.9236
rs7571606_A  0.4618362            0 6.019053e-06 6.016629e-06 -569.5354
rs10803511_C 0.4622585            0 6.846865e-06 6.844186e-06 -566.0754
rs2138445_C  0.4620487            0 7.082199e-06 7.079448e-06 -565.4754
rs9287372_G  0.4620392            0 6.763720e-06 6.761066e-06 -566.7569
rs10427255_T 0.4620517            0 7.085743e-06 7.082990e-06 -565.5130
rs1949330_G  0.4619910            0 6.964304e-06 6.961588e-06 -565.6939
rs7421123_A  0.4619928            0 7.207146e-06 7.204356e-06 -564.7179
rs1516135_T  0.4620021            0 7.416861e-06 7.414009e-06 -563.8953
rs1996287_T  0.4619772            0 7.358687e-06 7.355851e-06 -564.0431
rs1996286_T  0.4619781            0 7.309687e-06 7.306866e-06 -564.2206
rs2381726_C  0.4619797            0 7.299379e-06 7.296562e-06 -564.2531
rs2138448_G  0.4619789            0 7.353206e-06 7.350372e-06 -564.0179
rs6756212_T  0.4618215            0 8.740699e-06 8.737455e-06 -558.8317
rs13393501_C 0.4823639            0 2.274611e-05 2.273959e-05 -534.0060
rs1516145_T  0.4823836            0 2.249026e-05 2.248380e-05 -534.3840
rs2381712_G  0.4825300            0 1.801863e-05 1.801317e-05 -540.2935
rs2063862_G  0.4823982            0 2.334894e-05 2.334229e-05 -533.3349
rs34338764_T 0.4822215            0 2.434462e-05 2.433776e-05 -532.1429
rs4662420_A  0.4823360            0 2.303219e-05 2.302561e-05 -533.8011
rs1516141_G  0.4823333            0 2.282209e-05 2.281556e-05 -534.2390
rs10175039_T 0.4823268            0 2.268138e-05 2.267488e-05 -534.4205
rs10201277_C 0.4823297            0 2.292250e-05 2.291594e-05 -534.1428
rs10204857_G 0.4823336            0 2.206225e-05 2.205588e-05 -535.1424
rs6745444_G  0.4823034            0 2.040591e-05 2.039991e-05 -537.1892
rs6721116_A  0.4822490            0 2.032906e-05 2.032307e-05 -537.2026
rs12617922_A 0.4822474            0 1.963027e-05 1.962444e-05 -538.0827
                  Var         z           
rs961414_T   15816.20 -4.381086 0.02430087
rs7596680_G  15817.84 -4.386190 0.02475122
rs4422110_T  15822.23 -4.372915 0.02348297
rs1474011_A  15837.26 -4.502008 0.04117635
rs1533426_A  15832.36 -4.521821 0.04473008
rs6430080_A  15834.11 -4.520643 0.04449596
rs13009175_T 15843.76 -4.496025 0.04018651
rs7571606_A  15836.16 -4.525804 0.04544534
rs10803511_C 15834.98 -4.498477 0.04065538
rs2138445_C  15852.08 -4.491286 0.03925386
rs9287372_G  15854.81 -4.501075 0.04089321
rs10427255_T 15854.93 -4.491179 0.03923206
rs1949330_G  15839.11 -4.494860 0.03991979
rs7421123_A  15835.89 -4.487560 0.03874554
rs1516135_T  15832.93 -4.481443 0.03777104
rs1996287_T  15829.36 -4.483123 0.03807728
rs1996286_T  15829.26 -4.484548 0.03832149
rs2381726_C  15828.96 -4.484849 0.03837502
rs2138448_G  15826.82 -4.483282 0.03812738
rs6756212_T  15796.85 -4.446269 0.03278549
rs13393501_C 15891.02 -4.236139 0.02002611
rs1516145_T  15894.45 -4.238680 0.02016356
rs2381712_G  15874.89 -4.288194 0.02468269
rs2063862_G  15895.21 -4.230258 0.01952405
rs34338764_T 15894.82 -4.220855 0.01872521
rs4662420_A  15899.91 -4.233330 0.01976524
rs1516141_G  15910.52 -4.235390 0.01990042
rs10175039_T 15910.89 -4.236779 0.02001044
rs10201277_C 15912.21 -4.234403 0.01982594
rs10204857_G 15907.21 -4.242993 0.02052037
rs6745444_G  15897.91 -4.260468 0.02197360
rs6721116_A  15892.41 -4.261312 0.02206268
rs12617922_A 15886.24 -4.269121 0.02277305</code></pre>
<pre class="r"><code>rm(res, gwas, X, fit)</code></pre>
</div>
<div id="coa-2" class="section level4">
<h4>2. COA</h4>
<pre class="r"><code>region = &quot;chr2_143400001_147900000&quot;
res = readRDS(paste0(&quot;/Users/nicholeyang/Downloads/survivalsusie/result/asthma_self_report/result/coa/fit.susie.&quot;, region, &quot;.rds&quot;))
gwas = readRDS(paste0(&quot;/Users/nicholeyang/downloads/survivalsusie/result/gwas_surv/coa_gwas_&quot;, region, &quot;.rds&quot;))</code></pre>
<pre class="r"><code>fit = res[[1]]
X = res[[2]]

pip &lt;- logisticsusie:::get_pip(fit$alpha)
effect_estimate &lt;- colSums(fit$alpha * fit$mu)

class(fit) = &quot;susie&quot;
cs &lt;- susie_get_cs(fit, X)
cs</code></pre>
<pre><code>$cs
NULL

$coverage
NULL

$requested_coverage
[1] 0.95</code></pre>
<pre class="r"><code>rm(res, gwas, X, fit)</code></pre>
</div>
<div id="aoa-2" class="section level4">
<h4>3. AOA</h4>
<pre class="r"><code>region = &quot;chr2_143400001_147900000&quot;
res = readRDS(paste0(&quot;/Users/nicholeyang/Downloads/survivalsusie/result/asthma_self_report/result/aoa/fit.susie.&quot;, region, &quot;.rds&quot;))
gwas = readRDS(paste0(&quot;/Users/nicholeyang/downloads/survivalsusie/result/gwas_surv/aoa_gwas_&quot;, region, &quot;.rds&quot;))</code></pre>
<pre class="r"><code>fit = res[[1]]
X = res[[2]]

pip &lt;- logisticsusie:::get_pip(fit$alpha)
effect_estimate &lt;- colSums(fit$alpha * fit$mu)

class(fit) = &quot;susie&quot;
cs &lt;- susie_get_cs(fit, X)
cs</code></pre>
<pre><code>$cs
$cs$L1
 [1] 815 817 820 828 831 832 848 849 850 852 854 855 869 871 872 875 876 877 881
[20] 902 911 918 920 922 937 951 952 954 958 961 966 970


$purity
   min.abs.corr mean.abs.corr median.abs.corr
L1    0.9333614     0.9716784       0.9901346

$cs_index
[1] 1

$coverage
[1] 0.9573009

$requested_coverage
[1] 0.95</code></pre>
<pre class="r"><code>snps1 = colnames(X)[cs$cs$L1]
colors &lt;- ifelse(rownames(gwas) %in% snps1, &quot;red&quot;, &quot;black&quot;)
plot(-log10(gwas[, &quot;p.value.spa&quot;]), col = colors, xlab = &quot;SNP&quot;, ylab = &quot;-log10(p-value)&quot;, cex = 0.8, pch = 20, main = &quot;CS 1&quot;)</code></pre>
<p><img src="figure/susie_asthma_result2.Rmd/unnamed-chunk-42-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-42-1">
Past versions of unnamed-chunk-42-1.png
</button>
</p>
<div id="fig-unnamed-chunk-42-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/e5d84dd7238d4a04ec4d4133ce12fc5f76e152f1/docs/figure/susie_asthma_result2.Rmd/unnamed-chunk-42-1.png" target="_blank">e5d84dd</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cbind(gwas[rownames(gwas) %in% snps1, ], pip[sort(cs$cs$L1)])</code></pre>
<pre><code>                   MAF missing.rate  p.value.spa p.value.norm      Stat
rs961414_T   0.4667480            0 3.587785e-09 3.575483e-09 -567.2621
rs7596680_G  0.4667207            0 3.327163e-09 3.315650e-09 -568.4859
rs4422110_T  0.4668022            0 3.585143e-09 3.572852e-09 -567.3869
rs1474011_A  0.4620587            0 3.104126e-09 3.093126e-09 -569.9077
rs1533426_A  0.4617573            0 2.761546e-09 2.751611e-09 -571.6736
rs6430080_A  0.4617327            0 2.822795e-09 2.812667e-09 -571.3597
rs13009175_T 0.4620295            0 3.222931e-09 3.211561e-09 -569.4308
rs7571606_A  0.4617688            0 2.702730e-09 2.692982e-09 -572.0808
rs10803511_C 0.4621893            0 3.194542e-09 3.183262e-09 -569.4129
rs2138445_C  0.4619782            0 3.213446e-09 3.202104e-09 -569.6267
rs9287372_G  0.4619688            0 3.191584e-09 3.180309e-09 -569.7842
rs10427255_T 0.4619800            0 3.335727e-09 3.324006e-09 -569.0889
rs1949330_G  0.4619177            0 3.084471e-09 3.073529e-09 -570.0443
rs7421123_A  0.4619175            0 3.037673e-09 3.026878e-09 -570.2262
rs1516135_T  0.4619258            0 3.096824e-09 3.085844e-09 -569.8690
rs1996287_T  0.4619006            0 3.030006e-09 3.019233e-09 -570.1497
rs1996286_T  0.4619016            0 3.004928e-09 2.994233e-09 -570.2794
rs2381726_C  0.4619032            0 2.997264e-09 2.986593e-09 -570.3143
rs2138448_G  0.4619020            0 2.993832e-09 2.983173e-09 -570.2948
rs6756212_T  0.4617370            0 3.049255e-09 3.038412e-09 -569.4825
rs13393501_C 0.4822528            0 5.929335e-09 5.910926e-09 -560.5937
rs1516145_T  0.4822710            0 5.583078e-09 5.565618e-09 -561.6246
rs2381712_G  0.4824146            0 3.529489e-09 3.517830e-09 -568.6085
rs2063862_G  0.4822842            0 5.736962e-09 5.719080e-09 -561.2017
rs4662420_A  0.4822219            0 5.579847e-09 5.562396e-09 -561.7330
rs1516141_G  0.4822184            0 5.321629e-09 5.304891e-09 -562.6852
rs10175039_T 0.4822115            0 5.211446e-09 5.195013e-09 -563.0280
rs10201277_C 0.4822145            0 5.324454e-09 5.307708e-09 -562.7069
rs10204857_G 0.4822188            0 5.006536e-09 4.990674e-09 -563.6046
rs6745444_G  0.4821912            0 5.056727e-09 5.040721e-09 -563.2797
rs6721116_A  0.4821369            0 5.146031e-09 5.129776e-09 -562.8979
rs12617922_A 0.4821366            0 5.038611e-09 5.022653e-09 -563.1278
                  Var         z           
rs961414_T   9235.545 -5.902724 0.03077770
rs7596680_G  9236.497 -5.915154 0.03289158
rs4422110_T  9239.232 -5.902845 0.03081744
rs1474011_A  9247.003 -5.926578 0.03571079
rs1533426_A  9244.431 -5.945769 0.03966456
rs6430080_A  9245.459 -5.942174 0.03889306
rs13009175_T 9250.805 -5.920402 0.03455088
rs7571606_A  9246.632 -5.949296 0.04043099
rs10803511_C 9245.679 -5.921857 0.03482715
rs2138445_C  9255.655 -5.920887 0.03465308
rs9287372_G  9257.262 -5.922010 0.03485663
rs10427255_T 9257.397 -5.914740 0.03352549
rs1949330_G  9248.177 -5.927622 0.03593673
rs7421123_A  9246.242 -5.930134 0.03642080
rs1516135_T  9244.540 -5.926965 0.03579409
rs1996287_T  9242.470 -5.930549 0.03651092
rs1996286_T  9242.418 -5.931914 0.03677726
rs2381726_C  9242.245 -5.932333 0.03686159
rs2138448_G  9241.025 -5.932522 0.03689379
rs6756212_T  9224.086 -5.929509 0.03623298
rs13393501_C 9280.280 -5.819259 0.02361723
rs1516145_T  9282.343 -5.829313 0.02487404
rs2381712_G  9271.020 -5.905404 0.03712859
rs2063862_G  9282.826 -5.824771 0.02430319
rs4662420_A  9285.619 -5.829409 0.02487356
rs1516141_G  9291.907 -5.837314 0.02590142
rs10175039_T 9292.123 -5.840802 0.02636998
rs10201277_C 9292.906 -5.837226 0.02588728
rs10204857_G 9289.902 -5.847483 0.02730284
rs6745444_G  9284.468 -5.845822 0.02709423
rs6721116_A  9281.142 -5.842907 0.02670461
rs12617922_A 9277.564 -5.846420 0.02721545</code></pre>
<pre class="r"><code>rm(res, gwas, X, fit)</code></pre>
</div>
</div>
<div id="region-4" class="section level3">
<h3>Region 4</h3>
<p>Both very significant signals for AOA and COA, pval = 1e-20.</p>
<div id="all-asthma-cases-3" class="section level4">
<h4>1. All asthma cases</h4>
<pre class="r"><code>region = &quot;chr6_30500001_32100000&quot;
res = readRDS(paste0(&quot;/Users/nicholeyang/Downloads/survivalsusie/result/asthma_self_report/result/all/fit.susie.&quot;, region, &quot;.rds&quot;))
gwas = readRDS(paste0(&quot;/Users/nicholeyang/downloads/survivalsusie/result/gwas_surv/all_gwas_&quot;, region, &quot;.rds&quot;))</code></pre>
<pre class="r"><code>fit = res[[1]]
X = res[[2]]
print(res[[3]])</code></pre>
<pre><code>    user   system  elapsed 
307736.2 114117.0  53094.4 </code></pre>
<pre class="r"><code>pip &lt;- logisticsusie:::get_pip(fit$alpha)
effect_estimate &lt;- colSums(fit$alpha * fit$mu)
pip.sorted = sort(pip, decreasing = TRUE)
pip.sorted[1:10]</code></pre>
<pre><code> [1] 1.00000000 0.42929209 0.40183240 0.05935401 0.05427636 0.05280927
 [7] 0.05210378 0.04896134 0.04692130 0.04571521</code></pre>
<pre class="r"><code>class(fit) = &quot;susie&quot;
cs &lt;- susie_get_cs(fit, X)
cs</code></pre>
<pre><code>$cs
$cs$L1
[1] 2501

$cs$L2
[1]   58 1262 1421 2056 2123 2136


$purity
   min.abs.corr mean.abs.corr median.abs.corr
L1    1.0000000     1.0000000       1.0000000
L2    0.8454127     0.9208507       0.9038682

$cs_index
[1] 1 2

$coverage
[1] 1.0000000 0.9522076

$requested_coverage
[1] 0.95</code></pre>
<pre class="r"><code>par(mfrow = c(1,2))
snps1 = colnames(X)[cs$cs$L1]
colors &lt;- ifelse(rownames(gwas) %in% snps1, &quot;red&quot;, &quot;black&quot;)
plot(-log10(gwas[, &quot;p.value.spa&quot;]), col = colors, xlab = &quot;SNP&quot;, ylab = &quot;-log10(p-value)&quot;, cex = 0.8, pch = 20, main = &quot;CS 1&quot;)

snps2 = colnames(X)[cs$cs$L2]
colors &lt;- ifelse(rownames(gwas) %in% snps2, &quot;red&quot;, &quot;black&quot;)
plot(-log10(gwas[, &quot;p.value.spa&quot;]), col = colors, xlab = &quot;SNP&quot;, ylab = &quot;-log10(p-value)&quot;, cex = 0.8, pch = 20, main = &quot;CS 2&quot;)</code></pre>
<p><img src="figure/susie_asthma_result2.Rmd/unnamed-chunk-47-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-47-1">
Past versions of unnamed-chunk-47-1.png
</button>
</p>
<div id="fig-unnamed-chunk-47-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/e5d84dd7238d4a04ec4d4133ce12fc5f76e152f1/docs/figure/susie_asthma_result2.Rmd/unnamed-chunk-47-1.png" target="_blank">e5d84dd</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>print(snps1)</code></pre>
<pre><code>[1] &quot;rs2428494_A&quot;</code></pre>
<pre class="r"><code>cbind(gwas[rownames(gwas) %in% snps1, ], pip[sort(cs$cs$L1)])</code></pre>
<pre><code>                     [,1] [,2]
MAF          4.770845e-01    1
missing.rate 0.000000e+00    1
p.value.spa  2.359694e-51    1
p.value.norm 2.196762e-51    1
Stat         1.894944e+03    1
Var          1.579054e+04    1
z            1.507988e+01    1</code></pre>
<pre class="r"><code>cbind(gwas[rownames(gwas) %in% snps2, ], pip[sort(cs$cs$L2)])</code></pre>
<pre><code>                     MAF missing.rate  p.value.spa p.value.norm      Stat
rs4713451_C   0.08872864            0 3.316825e-13 2.852047e-13 -522.9695
rs9468965_A   0.07710370            0 1.821596e-13 1.512565e-13 -457.4414
rs114444221_G 0.06422248            0 2.759637e-12 2.328671e-12 -375.7347
rs113169753_A 0.08208212            0 1.052982e-11 9.428462e-12 -466.3420
rs113511111_A 0.08211077            0 1.016966e-11 9.101730e-12 -466.8795
rs111606016_C 0.08202514            0 1.144260e-11 1.025736e-11 -465.6304
                   Var         z           
rs4713451_C   5130.548 -7.301204 0.40183240
rs9468965_A   3835.717 -7.386049 0.42929209
rs114444221_G 2870.289 -7.013237 0.05427636
rs113169753_A 4682.540 -6.814968 0.02484358
rs113511111_A 4686.365 -6.820037 0.02549773
rs111606016_C 4684.914 -6.802844 0.02331646</code></pre>
</div>
<div id="conditional-analysis-1" class="section level4">
<h4>2. Conditional analysis</h4>
<pre class="r"><code>gwas1 &lt;- readRDS(&quot;/Users/nicholeyang/Downloads/survivalsusie/result/asthma_self_report/result/gwas_surv_conditional/all_gwas_chr6_30500001_32100000_rs2428494_A.rds&quot;)</code></pre>
<pre class="r"><code>snps2 = colnames(X)[cs$cs$L2]
colors &lt;- ifelse(rownames(gwas1) %in% snps2, &quot;red&quot;, &quot;black&quot;)
plot(-log10(gwas1[, &quot;p.value.spa&quot;]), col = colors, xlab = &quot;SNP&quot;, ylab = &quot;-log10(p-value)&quot;, cex = 0.8, pch = 20, main = &quot;CS 2&quot;)</code></pre>
<p><img src="figure/susie_asthma_result2.Rmd/unnamed-chunk-50-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-50-1">
Past versions of unnamed-chunk-50-1.png
</button>
</p>
<div id="fig-unnamed-chunk-50-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/e5d84dd7238d4a04ec4d4133ce12fc5f76e152f1/docs/figure/susie_asthma_result2.Rmd/unnamed-chunk-50-1.png" target="_blank">e5d84dd</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/b7003474a5621d0b2505affe593782361b83f730/docs/figure/susie_asthma_result2.Rmd/unnamed-chunk-50-1.png" target="_blank">b700347</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>rm(res, gwas, X, fit)</code></pre>
</div>
<div id="coa-3" class="section level4">
<h4>2. COA</h4>
<pre class="r"><code>region = &quot;chr6_30500001_32100000&quot;
res = readRDS(paste0(&quot;/Users/nicholeyang/Downloads/survivalsusie/result/asthma_self_report/result/coa/fit.susie.&quot;, region, &quot;.rds&quot;))
gwas = readRDS(paste0(&quot;/Users/nicholeyang/downloads/survivalsusie/result/gwas_surv/coa_gwas_&quot;, region, &quot;.rds&quot;))</code></pre>
<pre class="r"><code>fit = res[[1]]
X = res[[2]]

pip &lt;- logisticsusie:::get_pip(fit$alpha)
effect_estimate &lt;- colSums(fit$alpha * fit$mu)

class(fit) = &quot;susie&quot;
cs &lt;- susie_get_cs(fit, X)
cs</code></pre>
<pre><code>$cs
$cs$L1
[1] 2501


$purity
   min.abs.corr mean.abs.corr median.abs.corr
L1            1             1               1

$cs_index
[1] 1

$coverage
[1] 0.9999816

$requested_coverage
[1] 0.95</code></pre>
<pre class="r"><code>snps1 = colnames(X)[cs$cs$L1]
colors &lt;- ifelse(rownames(gwas) %in% snps1, &quot;red&quot;, &quot;black&quot;)
plot(-log10(gwas[, &quot;p.value.spa&quot;]), col = colors, xlab = &quot;SNP&quot;, ylab = &quot;-log10(p-value)&quot;, cex = 0.8, pch = 20, main = &quot;CS 1&quot;)</code></pre>
<p><img src="figure/susie_asthma_result2.Rmd/unnamed-chunk-54-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-54-1">
Past versions of unnamed-chunk-54-1.png
</button>
</p>
<div id="fig-unnamed-chunk-54-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/e5d84dd7238d4a04ec4d4133ce12fc5f76e152f1/docs/figure/susie_asthma_result2.Rmd/unnamed-chunk-54-1.png" target="_blank">e5d84dd</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/b7003474a5621d0b2505affe593782361b83f730/docs/figure/susie_asthma_result2.Rmd/unnamed-chunk-54-1.png" target="_blank">b700347</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cbind(gwas[rownames(gwas) %in% snps1, ], pip[sort(cs$cs$L1)])</code></pre>
<pre><code>                     [,1]      [,2]
MAF          4.764637e-01 0.9999816
missing.rate 0.000000e+00 0.9999816
p.value.spa  1.580844e-23 0.9999816
p.value.norm 1.441063e-23 0.9999816
Stat         6.312377e+02 0.9999816
Var          3.980201e+03 0.9999816
z            1.000554e+01 0.9999816</code></pre>
<pre class="r"><code>rm(res, gwas, X, fit)</code></pre>
</div>
<div id="aoa-3" class="section level4">
<h4>3. AOA</h4>
<pre class="r"><code>region = &quot;chr6_30500001_32100000&quot;
res = readRDS(paste0(&quot;/Users/nicholeyang/Downloads/survivalsusie/result/asthma_self_report/result/aoa/fit.susie.&quot;, region, &quot;.rds&quot;))
gwas = readRDS(paste0(&quot;/Users/nicholeyang/downloads/survivalsusie/result/gwas_surv/aoa_gwas_&quot;, region, &quot;.rds&quot;))</code></pre>
<pre class="r"><code>fit = res[[1]]
X = res[[2]]

pip &lt;- logisticsusie:::get_pip(fit$alpha)
effect_estimate &lt;- colSums(fit$alpha * fit$mu)

class(fit) = &quot;susie&quot;
cs &lt;- susie_get_cs(fit, X)
cs</code></pre>
<pre><code>$cs
$cs$L1
[1] 2501


$purity
   min.abs.corr mean.abs.corr median.abs.corr
L1            1             1               1

$cs_index
[1] 1

$coverage
[1] 0.9999593

$requested_coverage
[1] 0.95</code></pre>
<pre class="r"><code>snps1 = colnames(X)[cs$cs$L1]
colors &lt;- ifelse(rownames(gwas) %in% snps1, &quot;red&quot;, &quot;black&quot;)
plot(-log10(gwas[, &quot;p.value.spa&quot;]), col = colors, xlab = &quot;SNP&quot;, ylab = &quot;-log10(p-value)&quot;, cex = 0.8, pch = 20, main = &quot;CS 1&quot;)</code></pre>
<p><img src="figure/susie_asthma_result2.Rmd/unnamed-chunk-58-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-58-1">
Past versions of unnamed-chunk-58-1.png
</button>
</p>
<div id="fig-unnamed-chunk-58-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-data-analysis/blob/e5d84dd7238d4a04ec4d4133ce12fc5f76e152f1/docs/figure/susie_asthma_result2.Rmd/unnamed-chunk-58-1.png" target="_blank">e5d84dd</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cbind(gwas[rownames(gwas) %in% snps1, ], pip[sort(cs$cs$L1)])</code></pre>
<pre><code>                     [,1]      [,2]
MAF          4.753661e-01 0.9999593
missing.rate 0.000000e+00 0.9999593
p.value.spa  1.519018e-23 0.9999593
p.value.norm 1.472299e-23 0.9999593
Stat         9.597374e+02 0.9999593
Var          9.204670e+03 0.9999593
z            1.000342e+01 0.9999593</code></pre>
<pre class="r"><code>rm(res, gwas, X, fit)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin20.6.0 (64-bit)
Running under: macOS Monterey 12.0.1

Matrix products: default
BLAS:   /usr/local/Cellar/openblas/0.3.18/lib/libopenblasp-r0.3.18.dylib
LAPACK: /usr/local/Cellar/r/4.1.1_1/lib/R/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] logisticsusie_0.0.0.9004 testthat_3.1.0           susieR_0.12.35          
[4] survival_3.2-11          workflowr_1.6.2         

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3       lattice_0.20-44    prettyunits_1.1.1  ps_1.6.0          
 [5] rprojroot_2.0.2    digest_0.6.28      utf8_1.2.2         R6_2.5.1          
 [9] plyr_1.8.6         RcppZiggurat_0.1.6 evaluate_0.14      highr_0.9         
[13] ggplot2_3.4.3      pillar_1.9.0       rlang_1.1.1        rstudioapi_0.13   
[17] irlba_2.3.5        whisker_0.4        callr_3.7.3        jquerylib_0.1.4   
[21] Matrix_1.5-3       rmarkdown_2.11     desc_1.4.0         devtools_2.4.2    
[25] splines_4.1.1      stringr_1.4.0      munsell_0.5.0      mixsqp_0.3-43     
[29] compiler_4.1.1     httpuv_1.6.3       xfun_0.27          pkgconfig_2.0.3   
[33] pkgbuild_1.2.0     htmltools_0.5.5    tidyselect_1.2.0   tibble_3.1.5      
[37] matrixStats_0.63.0 reshape_0.8.9      fansi_0.5.0        crayon_1.4.1      
[41] dplyr_1.0.7        withr_2.5.0        later_1.3.0        grid_4.1.1        
[45] jsonlite_1.7.2     gtable_0.3.0       lifecycle_1.0.3    git2r_0.28.0      
[49] magrittr_2.0.1     scales_1.2.1       Rfast_2.0.6        cli_3.6.1         
[53] stringi_1.7.5      cachem_1.0.6       fs_1.5.0           promises_1.2.0.1  
[57] remotes_2.4.2      bslib_0.4.1        ellipsis_0.3.2     generics_0.1.2    
[61] vctrs_0.6.3        tools_4.1.1        glue_1.4.2         purrr_0.3.4       
[65] parallel_4.1.1     processx_3.8.1     pkgload_1.2.3      fastmap_1.1.0     
[69] yaml_2.2.1         colorspace_2.0-2   sessioninfo_1.1.1  memoise_2.0.1     
[73] knitr_1.36         usethis_2.1.3      sass_0.4.4        </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
